scripts/version.js

```diff
@@ -8,30 +8,34 @@ const title = require('./helpers/title')
 
 const pkg = require('../package.json')
 const increment = getIncrement()
-const version = semver.inc(pkg.version, increment)
-const branch = `v${semver.major(version)}.${semver.minor(version)}`
-const tag = `v${version}`
-const isNewBranch = semver.major(pkg.version) !== semver.major(version) ||
-  semver.minor(pkg.version) !== semver.minor(version)
-
-title(`Bumping version to v${version} in a new branch`)
-
-pkg.version = version
-
-if (isNewBranch) {
-  exec(`git pull`)
-  exec(`git checkout -b ${branch}`)
-} else {
-  exec(`git checkout ${branch}`)
-  exec('git pull')
+const version = semver.inc(pkg.version, increment, 'pre')
+
+title(`Bumping version to v${version}.`)
+
+const currentBranch = exec.pipe(`git branch --show-current`)
+
+if (currentBranch === 'master') {
+  exec(`git checkout -b v${semver.major(pkg.version)}.x`)
+  exec(`git push -u origin HEAD`)
+
+  const nextMajor = semver.major(pkg.version) + 1
+
+  bump(`v${nextMajor}.0.0-pre`)
```

Comment:
I think you need to either return here or put the other bump in an else?

Reply:
For `master`, both bumps need to happen. One is the version bump for the release branch, and the other is the version bump for master.